### DISCUSSION TITLE
Is there anyway to get namePath from a Form.Item in a child component?

### DISCUSSION THREAD

**Discussion by yhy-1** [ant-design/ant-design#45239] [General]
Hi, I am writing a HOC component and would like to validate a form item without having to pass the namePath from form.Item to the child component. 

```
const MyCustomComponent = (props) => {
 const form = Form.useFormInstance();
 const onChange= ()=>{
    form.validateFields("Some kind of way to get the namePath?")
 }

  <div>
    <h3>I am a wrapped Input</h3>
    <MyCustomComponent onChange={onChange} {...props} />
  </div>
};

```
// user will pass these 
```
<Form.Item  
    rules={[{
                required: true,
        }]} 
    name="input">
  <MyCustomComponent />
</Form.Item>;
```

**Comment by Yuiai01** [comment]
Maybe you can try using the [getFieldsValue](https://ant.design/components/form-cn#forminstance).

**Reply by yhy-1** [reply]
Still need to pass the field name to getFieldsValue()

**Reply by Yuiai01** [reply]
I'm sorry I didn't make it clear, I meant to say you can use `getFieldsValue()` to get all filed name.

**Reply by yhy-1** [reply]
I only want to validate the current form item. Is there any way to know from all the file names the current field name?

**Reply by Yuiai01** [reply]
There is currently no way to get a namePath in this way, but you can refer to [Customized Form Controls](https://ant.design/components/form#components-form-demo-customized-form-controls) and verify the field externally.

**Comment by crazyair** [comment]
`MyCustomComponent` where to use it?

**Reply by yhy-1** [reply]
Sorry, I updated to code. Look like it wasn't properly showing.

**Comment by crazyair** [comment]
Validate in Form.Item, don't in Component
```tsx
<Form.Item rulles={[{ ...validate }]}>
```
